AGENTS.md

```diff
@@ -0,0 +1,952 @@
+
+- **Purpose**: Provide reusable code for common operations
+- **Location**: `robottelo/`
+- **Components**:
+  - **API helpers**: `robottelo/api/` (Nailgun entities)
+  - **CLI helpers**: `robottelo/cli/` (Hammer command wrappers)
+  - **Host helpers**: `robottelo/host_helpers/` (Satellite/ContentHost mixins)
+  - **Utilities**: `robottelo/utils/` (decorators, data factories, etc.)
+
+### Layer 4: Infrastructure Layer
+External services and tools that tests depend on.
+
+- **Broker**: VM/host provisioning
+- **Manifester**: Subscription manifest generation
+- **Report Portal**: Test result reporting
+- **Vault**: Secret management
+
+---
+
+## Key Concepts
+
+### 1. **Fixtures**
+
+Pytest fixtures provide test dependencies and setup/teardown logic.
+
+**Core Fixtures**:
+- `target_sat`: A Satellite instance for the test
+- `module_target_sat`: Module-scoped Satellite instance
+- `function_sca_manifest`: Function-scoped SCA manifest
+- `rhel_contenthost`: RHEL content host for testing
+
+**Component Fixtures**:
+- `module_ak_with_cv`: Activation key with content view
+- `module_lce`: Lifecycle environment
+- `module_org`: Organization
+
+**Fixture Scopes**:
+- `function`: Per test function (default)
+- `module`: Per test module
+- `session`: Per test session
+
+Example:
+
+```python
+@pytest.fixture
+def activation_key(module_org, module_target_sat):
+    """Create an activation key for testing"""
+    ak = module_target_sat.api.ActivationKey(
+        organization=module_org,
+        name='test-ak'
+    ).create()
+    yield ak
+```
+
+### 2. **Markers**
+
+Pytest markers categorize and filter tests.
+
+**Common Markers**:
+- `@pytest.mark.tier1`: High priority tests
+- `@pytest.mark.tier2`: Medium priority tests
+- `@pytest.mark.tier3`: Low priority tests
+- `@pytest.mark.e2e`: End-to-end workflow tests
+- `@pytest.mark.rhel_ver_match()`: Filter by RHEL version
+- `@pytest.mark.parametrize()`: Parameterize test inputs
+
+Example:
+
+```python
+@pytest.mark.tier1
+@pytest.mark.rhel_ver_match(r'^(9|10)')
+def test_positive_create_ak(module_org, module_target_sat):
+    """Test activation key creation"""
+    ak = module_target_sat.api.ActivationKey(
+        organization=module_org
+    ).create()
+```
+
+### 3. **Host Helpers**
+
+Mixins that provide common functionality for Satellite and ContentHost objects.
+
+**Satellite Mixins** (`robottelo/host_helpers/satellite_mixins.py`):
+- `api_factory`: API entity creation methods
+- `cli_factory`: CLI entity creation methods
+- `ui_session()`: Context manager for UI sessions
+
+**ContentHost Mixins** (`robottelo/host_helpers/contenthost_mixins.py`):
+- `register_contenthost()`: Register host to Satellite
+- `install_katello_ca()`: Install Katello CA certificate
+- `execute()`: Run commands on content host
+
+Example:
+
+```python
+# Using Satellite API factory
+ak = target_sat.api.ActivationKey(organization=org).create()
+
+# Using UI session
+with target_sat.ui_session() as session:
```

Comment:
I find it a good practice to do Org/Loc selection as first step in the UI tests.
```suggestion
with target_sat.ui_session() as session:
	session.organization.select('ORG_NAME')
	session.location.select('LOC_NAME')
```